cips/cip-X.md

```diff
@@ -0,0 +1,271 @@
+---
+cip: <to be assigned>
+title: Off-chain Messages Signing
+author: @JulianToledano
+status: Draft
+type: Standards Track
+created: 2023-10-10
+---
+
+## Simple Summary
+
+This is a convention for signing arbitrary messages. We propose this specification as a method for signing and
+validating off-chain arbitrary messages.
+
+## Abstract
+
+This CIP addresses the need for arbitrary message signing. These messages can be used for application sign-in by
+providing proof of wallet ownership, but their use is not limited to this case. They can also be utilized for signing
+files, text, and more.
+
+There is no requirement for these messages to be posted to the network. That's why a set of off-chain messages is
+defined with specific characteristics that prevent them from being processed as on-chain transactions.
+
+This CIP defines two different messages. The first one allows users to sign any arbitrary data, including files, text,
+objects, and more. The second one provides a way for application sign-in by defining a standard human-readable message
+format that is parameterized by scope, session details, and security mechanisms.
+
+## Motivation
+
+Having the ability to sign messages off-chain has proven to be a fundamental aspect of nearly any blockchain.
+The notion of signing messages off-chain has many added benefits, such as saving on computational costs and reducing
+transaction throughput and overhead. Within the context of the Cosmos ecosystem, some of the major applications of signing such
+data include, but are not limited to, providing a cryptographically secure and verifiable means of proving validator
+identity and possibly associating it with some other framework or organization. Additionally, having the ability to
+sign Cosmos messages with a Ledger or similar HSM device.
```

Comment:
You could add smart accounts here, where off-chain signing allows for application layer authorizations of something.

E.g. we have a CosmWasm smart contract that verifies Ethereum arbitrary messages in a contract for a long time. Similarly folks now request secp256r1 message verification in CosmWasm for the same use case: some keypair signs arbitrary data off-chain and the contract can verify it.